pkg/intermediate/aggregate.go

```diff
@@ -524,59 +548,101 @@ func (a *AggregationProcess) aggregateRecords(incomingRecord, existingRecord ent
+					existingVal := srcExistingIeWithValue.GetUnsigned64Value()
 					if !isDelta {
-						existingIeWithValue.SetUnsigned64Value(incomingVal)
+						srcExistingIeWithValue.SetUnsigned64Value(incomingVal)
+						switch antreaSourceStatsElements[i] {
+						case "octetTotalCountFromSourceNode":
+							totalCountDiff = incomingVal - existingVal
+						case "reverseOctetTotalCountFromSourceNode":
+							reverseTotalCountDiff = incomingVal - existingVal
+						}
 					} else {
-						existingIeWithValue.SetUnsigned64Value(incomingVal + existingIeWithValue.GetUnsigned64Value())
+						srcExistingIeWithValue.SetUnsigned64Value(incomingVal + existingVal)
 					}
 				} else {
 					return fmt.Errorf("element does not exist in the record: %v", antreaSourceStatsElements[i])
 				}
 			}
-			// Update the corresponding destination element in antreaStatsElement list.
+			// Update the destination fields in antreaDestinationStatsElements list
 			if fillDstStats {
-				if existingIeWithValue, _, exist = existingRecord.GetInfoElementWithValue(antreaDestinationStatsElements[i]); exist {
+				if dstExistingIeWithValue, _, exist := existingRecord.GetInfoElementWithValue(antreaDestinationStatsElements[i]); exist {
+					existingVal := dstExistingIeWithValue.GetUnsigned64Value()
 					if !isDelta {
-						existingIeWithValue.SetUnsigned64Value(incomingVal)
+						dstExistingIeWithValue.SetUnsigned64Value(incomingVal)
+						switch antreaDestinationStatsElements[i] {
+						case "octetTotalCountFromDestinationNode":
+							totalCountDiff = incomingVal - existingVal
+						case "reverseOctetTotalCountFromDestinationNode":
+							reverseTotalCountDiff = incomingVal - existingVal
+						}
 					} else {
-						existingIeWithValue.SetUnsigned64Value(incomingVal + existingIeWithValue.GetUnsigned64Value())
+						dstExistingIeWithValue.SetUnsigned64Value(incomingVal + existingVal)
 					}
 				} else {
 					return fmt.Errorf("element does not exist in the record: %v", antreaDestinationStatsElements[i])
 				}
 			}
+			// Update the corresponding common element in statsElement list.
+			commonExistingIeWithValue, _, _ := existingRecord.GetInfoElementWithValue(element)
+			if isLatest {
+				if !isDelta {
+					if commonExistingIeWithValue.GetUnsigned64Value() < incomingVal {
+						commonExistingIeWithValue.SetUnsigned64Value(incomingVal)
+					}
+				} else {
+					if fillSrcStats {
+						srcIe, _, _ := existingRecord.GetInfoElementWithValue(antreaSourceStatsElements[i])
+						commonExistingIeWithValue.SetUnsigned64Value(srcIe.GetUnsigned64Value())
+					}
+					if fillDstStats {
+						dstIe, _, _ := existingRecord.GetInfoElementWithValue(antreaDestinationStatsElements[i])
+						commonExistingIeWithValue.SetUnsigned64Value(dstIe.GetUnsigned64Value())
+					}
+				}
+			}
 		} else {
 			return fmt.Errorf("element with name %v in statsElements not present in the incoming record", element)
 		}
 	}
+
+	if flowEndSecondsDiff == 0 {
```

Comment:
although it's very unlikely - in the case when an agent is down and restarted, it will replay some flow records, when the first replayed record is exactly the one received before it's down.

I guess adding this zero check can avoid division by zero warning, in some extreme cases